pkg/flowaggregator/clickhouseclient/clickhouseclient.go

```diff
@@ -447,12 +406,23 @@ func ConnectClickHouse(url string) (*sql.DB, error) {
 	// Connect to ClickHouse in a loop
 	if err := wait.PollImmediate(connRetryInterval, connTimeout, func() (bool, error) {
 		// Open the database and ping it
-		var err error
-		connect, err = sql.Open("clickhouse", url)
-		if err != nil {
-			connErr = fmt.Errorf("error when opening DB connection: %v", err)
-			return false, nil
+		opt := clickhouse.Options{
+			Addr: []string{config.DatabaseURL},
+			Auth: clickhouse.Auth{
+				Username: config.Username,
+				Password: config.Password,
+				Database: config.Database,
+			},
+		}
+		var compression clickhouse.CompressionMethod
+		if *config.Compress {
+			compression = clickhouse.CompressionLZ4
 		}
+		opt.Compression = &clickhouse.Compression{
+			Method: compression,
+		}
+
+		connect = clickhouse.OpenDB(&opt)
```

Comment:
I may be wrong but it doesn't look like you are using the native ClickHouse interface here. This looks like `database/sql`, which would contradict your PR description.

https://github.com/ClickHouse/clickhouse-go#clickhouse-interface-formally-native-interface

Reply:
My bad. I will correct the PR comment. I chose `database/sql` API because it supports http protocol.